src/views/RegisterPage/index.js

```diff
@@ -27,10 +27,41 @@ class RegisterPage extends Component {
   }
 
   handleChange = event => {
-    this.setState({[event.target.name]: event.target.value});
+    if(event.target.name == "terms"){
+        if(event.target.checked){
+          this.setState({
+          [event.target.name]: false});
+        }
+        else{
+          this.setState({
+            [event.target.name]: true});
+        }
+      }
+    else{
+      this.setState({[event.target.name]: event.target.value});
+    }
   };
 
   handleSubmit = event => {
+    var error_flag = 0;   // if error_flag == 1, then an error in user input is detected.
+    if(this.state.email == null || this.state.firstname == null || this.state.lastname == null
```

Comment:
you can condense these two if statements by just doing one big one ->
`if(!this.state.email || this.state.email == "" || !this.state.firstname || this.state.firstname == "" || ... )`

Reply:
For some reason, when I do !this.state.email instead of this.state.email == null, the program does not behave properly. Like it would say "not all fields have been filled out", even though they are. For the time being, I have combined both the if statements into one big one, but left this.state.email == null